src/util/github.ts

```diff
@@ -1,47 +1,77 @@
+import { getInput } from '@actions/core'
+import { getOctokit } from '@actions/github'
+import { Context } from '@actions/github/lib/context'
+import { GitHub } from '@actions/github/lib/utils';
+import { ActionNotSupportedError, EventNotSupportedError, InvalidTokenError } from './github/errors';
+import { Event, PullRequestAction } from './github/events';
+import PullRequestHelper from './github/PullRequestHelper';
 
-export enum Event {
-  CHECK_RUN = 'check_run',
-  CHECK_SUITE = 'check_suite',
-  CREATE = 'create',
-  DELETE = 'delete',
-  DEPLOYMENT = 'deployment',
-  DEPLOYMENT_STATUS = 'deployment_status',
-  FORK = 'fork',
-  GOLLUM = 'gollum',
-  ISSUE_COMMENT = 'issue_comment',
-  ISSUES = 'issues',
-  LABEL = 'label',
-  MILESTONE = 'milestone',
-  PAGE_BUILD = 'page_build',
-  PROJECT = 'project',
-  PROJECT_CARD = 'project_card',
-  PROJECT_COLUMN = 'project_column',
-  PUBLIC = 'public',
-  PULL_REQUEST = 'pull_request',
-  PULL_REQUEST_REVIEW = 'pull_request_review',
-  PULL_REQUEST_REVIEW_COMMENT = 'pull_request_review_comment',
-  PULL_REQUEST_TARGET = 'pull_request_target',
-  PUSH = 'push',
-  REGISTRY_PACKAGE = 'registry_package',
-  RELEASE = 'release',
-  STATUS = 'status',
-  WATCH = 'watch',
-  WORKFLOW_RUN = 'workflow_run'
+const INPUT_TOKEN = 'token'
+
+const HELPER_BY_EVENT: Partial<Record<Event, EventHelperConstructors>> = {
+  [Event.PULL_REQUEST]: PullRequestHelper
 }
 
-export enum PullRequestAction {
-  ASSIGNED = 'assigned',
-  UNASSIGNED = 'unassigned',
-  LABELED = 'labeled',
-  UNLABELED = 'unlabeled',
-  OPENED = 'opened',
-  EDITED = 'edited',
-  CLOSED = 'closed',
-  REOPENED = 'reopened',
-  SYNCHRONIZE = 'synchronize',
-  READY_FOR_REVIEW = 'ready_for_review',
-  LOCKED = 'locked',
-  UNLOCKED = 'unlocked',
-  REVIEW_REQUESTED = 'review_requested',
-  REVIEW_REQUEST_REMOVED = 'review_request_removed'
+export type EventHelpers = PullRequestHelper | undefined
+export type EventHelperConstructors = typeof PullRequestHelper | undefined
+
+export type EventHandler = (context: Context, octokit: Octokit, helper?: PullRequestHelper) => void
+
+export type ActionHandlers = Partial<Record<string, Array<EventHandler>>>
+
+export type Octokit = InstanceType<typeof GitHub>
+
+export default class Github {
```

Comment:
Maybe this should be called `GithubActions`? It seems way more oriented around actions than Github in general.

Reply:
Also the filename has the wrong case, but a rename would help with this 😁

Reply:
Balls, wish I read this again before I noticed the case issue 🤦‍♂️